js/viewer/annotation/text.js

```diff
@@ -0,0 +1,425 @@
+    foreignObj.parentNode.removeChild(foreignObj);  
+}
+
+/**
+ * This function adds a textbox to the SVG group. It adds a foreignObject > div > textarea.
+ * If an importedObj exists, the function replaces the foreign object with the foreign object
+ * of imported annotation.
+ * @param {*} groupId 
+ * @param {*} importedObj - The foreign object from the imported annotation.
+ * @returns 
+ */
+Text.prototype.addTextBox = function (groupId, importedObj) {
+
+    // If a foreign object is to be imported from the loaded annotation
+    if(importedObj != null){
+        this.setForeignObj(importedObj);
+        var pTag = this.createPTag(importedObj.childNodes[0]);
+        importedObj.appendChild(pTag);
+        group = document.getElementById(groupId);
+        group.appendChild(importedObj);
+        group.setAttribute("contentEditable", "true");
+        return;
+    }
+
+    var svgForeignObj = document.createElementNS('http://www.w3.org/2000/svg', 'foreignObject')
+    var textOuterDiv = document.createElementNS("http://www.w3.org/1999/xhtml", "div");
+    var textInput = document.createElementNS("http://www.w3.org/1999/xhtml", "textarea");
+    this.setForeignObj(svgForeignObj);
+    
+    fontSize = document.getElementsByClassName("fontInput")[0].value;
+    this.changeFontSize(fontSize);
+
+    // Set the attributes of the foreign object, div and textarea
+    textInput.setAttribute("id", "textInput");
+    textOuterDiv.setAttribute("contentEditable", "true");
+    textInput.setAttribute("contentEditable", "true");
+    textInput.style.color = this._attrs["stroke"];
+    svgForeignObj.setAttribute("contentEditable", "true");
+    textOuterDiv.setAttribute("width", "auto");
+    textOuterDiv.setAttribute("tabindex", "-1");
+    svgForeignObj.setAttribute("width", "100%");
+    svgForeignObj.setAttribute("height", this._attrs["font-size"] * 100 + "px");
+    textOuterDiv.appendChild(textInput); 
+    svgForeignObj.classList.add("foreign-object");
+    textOuterDiv.classList.add("foreign-object-div");
+    svgForeignObj.setAttribute("x", this._attrs["x"]);
+    svgForeignObj.setAttribute("y", this._attrs["y"]);
+    svgForeignObj.appendChild(textOuterDiv);
+
+    group = document.getElementById(groupId);
+    group.appendChild(svgForeignObj);
+    group.setAttribute("contentEditable", "true");
+
+    // Add event listeners to the textarea for input
+    textInput.addEventListener("keydown", function (e) {
+        prev = this.textHeight;
+        this.style.height = (this.scrollHeight) + "px";
+        this.textHeight = this.scrollHeight;
+        textInput.innerHTML = textInput.value;
+    });
+
+    this.initResizeElement();
+    this.initDragElement();
+
+    textInput.style.height = this.textHeight + "px";
+    textInput.style.overflowY = "hidden";
+    textInput.style.fontSize = this._attrs["font-size"] + "px";
+    textOuterDiv.style.padding = (this.parent.getStrokeScale() * this._ratio * 4) + "px";
```

Comment:
Why *4?

Reply:
We want to create a draggable area for the div around the textarea. Added a comment in code as well